fl4health/server/base_server.py

```diff
@@ -43,16 +43,20 @@ def __init__(
             wandb_reporter (Optional[ServerWandBReporter], optional): To be provided if the server is to log
                 information and results to a Weights and Biases account. If None is provided, no logging occurs.
                 Defaults to None.
-            checkpointer (Optional[TorchCheckpointer], optional): To be provided if the server should perform
-                server side checkpointing based on some criteria. If none, then no server-side checkpointing is
-                performed. Defaults to None.
+            checkpointer (Optional[Union[TorchCheckpointer, Sequence
+                [TorchCheckpointer]]], optional): To be provided if the server
+                should perform server side checkpointing based on some
+                criteria. If none, then no server-side checkpointing is
+                performed. Multiple checkpointers can also be passed in a
+                sequence to checkpoint based on multiple criteria. Defaults to
+                None.
             metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
                 during the execution. Defaults to an instance of MetricsReporter with default init parameters.
         """
 
         super().__init__(client_manager=client_manager, strategy=strategy)
         self.wandb_reporter = wandb_reporter
-        self.checkpointer = checkpointer
+        self.checkpointer = [checkpointer] if isinstance(checkpointer, TorchCheckpointer) else checkpointer
```

Comment:
Should we assert that the names of the checkpointers are different to avoid users accidentally overwriting the checkpoints?

Reply:
Added clarification in comments